### ISSUE TITLE
Anytime data is loaded all the other input variables should be cleared.

### ISSUE THREAD

**Issue by moorepants** [ucd-ipo/aip-analysis#49]
- shinyjs can do this but it requires R > 3.1
- Here is a propsoal by the shinyjs author to add reset functionality to shiny: https://github.com/rstudio/shiny/issues/870

**Comment by daattali** [comment]
Is the R 3.1 version a restriction for you? What version do you have/can you not update R?

**Comment by moorepants** [comment]
The latest Ubuntu LTS release has R 3.0.2 and bunches of packages on CRAN fail to install with `install.packages()`. I'm newish to R packaging and found it really odd that all these packages don't support the R version in Ubuntu 14.04. I'm realizing that I'm going to be forced to use an external R install for things to work on Ubuntu, which in my opinion is odd and quite inconvenient. If you want my opinion, I'd say that R package authors should try to support R versions available in typical Linux distro installs and package managers. Thanks for checking in. I didn't realize if I cited your issue that it would ping you.

**Comment by daattali** [comment]
You're on ubuntu 14.04? You should be able to install the latest R version, you just need to set up the repos correctly. I recently wrote a tutorial on how to install the latest R on ubuntu 14.04 https://www.digitalocean.com/community/tutorials/how-to-set-up-r-on-ubuntu-14-04

Let me know if I'm misunderstanding the problem.  

BTW the reason shinyjs can't support older version of R is because shiny itself is a fairly new package and has had extensive work over the past few months,and doesn't work on very old R versions. And my package uses a lot of the newer shiny features

**Comment by moorepants** [comment]
Yes, I understand that I can install a newer version of R on Ubuntu 14.04 and I am coming to understand that is what R developers expect you do to instead of the other way around, i.e. the developers support a broader range of R versions, (especially those that are the default on the latest LTS release of the most popular Linux distro). This is just a growing pain for me as a new R developer. Seems that the onus is on the user to update instead of the developer to support recent past versions, which is different that the packaging ethos in other software communities I'm involved in.

Thanks for the link to your post!

**Comment by daattali** [comment]
Yes I can certainly agree with your pain. I'm fairly new to R too (haven't heard of it 2 years ago) and the first thing I was taught that in the R/Rstudio world updates are important.  To be honest, I'm seeing that both RStudio and R are getting new features and getting much better exponentially fast in the past 12 months, so I'm ok with that. I haven't seen any other language increase its usability so much in such short periods of time. But yes it's also a bit annoying that if you disappear off the earth for a year, when you come back stuff might not work anymore.

**Comment by moorepants** [comment]
Ok, the one bug associated with this issue is fixed from this simple fix: 6caae28

And after 3 hrs of goofing around it doesn't seem to be easy to reset all inputs in a general way. shinyjs doesn't support reseting inputs that are generated from renderUI, which are most of my ui elements. Shiny doesn't seem to want you to keep all the client side code in the ui file, i.e. it doesn't let you have access to client side input variables to drive dynamic ui generation all on the client.

Moving this out of the milestone as is it a lot of work for little gain to reset all inputs.

**Comment by daattali** [comment]
FYI if Shiny devs don't add support for this in the next few months, I'll do it myself and make a pull request to shiny. So in a few months you should be able to reset inputs if that's any help.
